src/Pages/ProductList/ProductList.js

```diff
@@ -1,7 +1,164 @@
-import React, { Component } from "react";
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import ProductCategory from "./Components/ProductCategory/ProductCategory";
+import ClassificationBox from "./Components/ClassificationBox/ClassificationBox";
+import { PRODUCTLIST_API } from "../../config";
+import { AiOutlineHome } from "react-icons/ai";
 
-export default class ProductList extends Component {
-  render() {
-    return <div></div>;
-  }
+export default function ProductList() {
+  const [categories, setCategories] = useState([]);
+  // const [newCategories, setNewCategories] = useState([]);
+  const [copyCategories, setCopyCategories] = useState([]);
+  const [state, setState] = useState({
+    seeAllCategories: "",
+    subcategoryCheckboxId1: "",
+    // subcategoryCheckboxId2: "",
+    // subcategoryCheckboxId3: "",
+  });
+  const [checkedNames, setCheckedNames] = useState(["전체 보기"]);
+
+  useEffect(() => {
+    // fetch(PRODUCTLIST_API)
+    fetch("/data/productList.json")
+      .then((res) => res.json())
+      .then((res) => setCategories(res.results));
+  }, []);
+
+  useEffect(() => {
+    setCopyCategories(categories);
+  }, [categories]);
+
+  // 1. [인풋 업데이트 (카테고리 이름 업뎃)] & 검사mtd(비동기)
+  const isCheckedCategoryName = (e) => {
+    const { name, value, checked } = e.target;
+    setState({ [name]: value });
+    makeCheckedNames(value);
+  };
+
+  // 2. [체크한 카테고리 배열에 넣는 mtd] (중복제거 mtd 포함)
+  const makeCheckedNames = (value) => {
+    const isInclude = checkedNames.some((name) => name === value);
+    const isIncludeAllBtn = checkedNames.some((name) => name === "전체 보기");
+
+    if (isInclude) {
+      const deleteOverlappedName = checkedNames.filter(
+        (name) => name !== value
+      );
+      setCheckedNames(deleteOverlappedName);
+    } else {
+      if (value === "전체 보기") {
+        setCheckedNames(["전체 보기"]);
+      } else {
+        if (isIncludeAllBtn) {
+          setCheckedNames([value]);
+          setCopyCategories([]);
+        } else {
+          setCheckedNames((prev) => [...prev, value]);
+        }
+      }
+    }
```

Comment:
여러 if문으로 작성해주시면 가독성에 더 좋습니다.
```suggestion
    if (isInclude) {
      const deleteOverlappedName = checkedNames.filter(
        (name) => name !== value
      );
      setCheckedNames(deleteOverlappedName);
    } 
    
    if (value === "전체 보기") {
      setCheckedNames(["전체 보기"]);
    } 

    if (isIncludeAllBtn) {
       setCheckedNames([value]);
       setCopyCategories([]);
    } 
        
    if (조건) {
      setCheckedNames((prev) => [...prev, value]);
    }
```